JavaScript1/Week2/homework.md

```diff
@@ -6,200 +6,157 @@
+```
+Now what happens if we call get status and activities is empty? We need to take that into consideration: If activities is empty log out a string that says: "Add some activities before calling ShowStatus"  
 
-``Today is: Sunday``    
-``How many days to meet : 9``  
-``We are meeting on : Tuesday``  
-
-
-
-Hint: ask each other for help/ask the right questions to the mentors.
-
-**2**  
-
-1. Write a JavaScript program
-    - Create an object of ``Students`` with properties: ``firstName``, ``lastName``, ``address``, ``telephone``, ``postCode``, ``course``, ``grade``.
-    - Create array of objects of students that reflects  your class. Give random grades to each students
-    anything between 60 and 95. Makes sure students names reflects your class mates. Assume everybody
-    lives in Copenhagen area.
-    - Create a function for finding average class grade.
-    - Create a function for finding student who has got the lowest grade. Put an encouraging message.
-    - Create a function for for finding student who got the highest grade. Put a celebration message.
+#### Usage limit
+A user asks us if it is possible to set a limit for his smartphone usage. "Off course it is" we promptly reply! 
+We need to store that limit somewhere, but where and what type should this be?
+So how should it work? When ShowStatus is called and the users usage is above the limit he set. Log out the following string: "You have reached your limit, no more smartphoning for you!"
 
-2. Write a JavaScript program which iterates the integers from 1 to 1000. But for multiples of three print "Fizz" instead of the number and for the multiples of five print "Buzz". 
-For numbers which are multiples of both three and five print "FizzBuzz".
+Come up with one feature you think would be helpful for this program.
 
-3. Write a JavaScript program to construct the following pattern, using a nested for loop. This might be bit of a challenge but read on nested for loop.
-    ````js
-        *
-        * *
-        * * *
-        * * * *
-        * * * * *
-        * * * * * *
-    ````
+Optional
+* Lets improve the addActivity, so that we dont need to specify the date, but the function automatically figures out what the date is. Check out this link: https://stackoverflow.com/a/34015511
+* Improve the ShowStatus function by only showing the number of actitivies for that specific day.
+* Create a function for calculating the activity a user has spent the most time on.
+---
 
 
-**3 (Challenge - only do this one once you have finished 1 and 2). Please work closely with your pair we identified in class.**
+## Step 4: Rover the Robot (Optional)
+Go and try out this cool game: [roverjs.com](http://roverjs.com), written by one of the HYF teachers from Amsterdam, Joost Lubach. There are different levels. Solve the chapter up until and with the variables chapter. There might be a small price for completing that!
```

Comment:
maybe we could link to Joost's github page so people can check his stuff out?

Reply:
Yeah, good idea :)